src/python/pants/backend/python/tasks/resolve_requirements_task_base.py

```diff
@@ -28,9 +31,17 @@ class ResolveRequirementsTaskBase(Task):
   @classmethod
   def prepare(cls, options, round_manager):
     round_manager.require_data(PythonInterpreter)
+    round_manager.require_data(BuildLocalPythonDistributions.PYTHON_DISTS)
 
   def resolve_requirements(self, req_libs):
-    with self.invalidated(req_libs) as invalidation_check:
+    """Requirements resolution for PEX files.
+
+    :param req_libs: A list of :class:`PythonRequirementLibrary` targets to resolve.
+    :returns: a PEX containing target requirements and any specified python dist targets.
+    """
+    local_dist_targets = self.context.targets(is_local_python_dist)
+    tgts = req_libs + local_dist_targets
+    with self.invalidated(tgts) as invalidation_check:
```

Comment:
I'd expect this to conform more to the existing calling convention - so ideally the caller passes in `local_dist_targets` as either a combined or separate param here? that would align better with how things are used below too, afaict?

Reply:
✅

Reply:
```tgts = req_libs + (local_dist_targets or [])``` works as well.